include/zephyr/platform/hooks.h

```diff
@@ -78,4 +78,12 @@ void board_early_init_hook(void);
  */
 void board_late_init_hook(void);
 
+/**
+ * @brief Hook executed when idle is entered.
+ *
+ * This hook is implemented by the SoC and can be used to perform any
+ * SoC-specific idle enter logic.
+ */
+void idle_enter_hook(void);
```

Comment:
If it's SoC specific, maybe clearer and more consistent is

```suggestion
void soc_idle_enter_hook(void);
```

Reply:
I have no problem with naming it soc_idle_enter_hook, it was my first choice as well but since its in idle.c and the Kconfig for it is IDLE_HOOK_ENTER, it felt more concrete to just name it idle :) I have no strong opinion though (well, between adding soc_ or not)